dash/development/base_component.py

```diff
@@ -167,11 +167,38 @@ def __init__(self, **kwargs):
 
             setattr(self, k, v)
 
-    def set_random_id(self):
-        if not hasattr(self, "id"):
-            v = str(uuid.UUID(int=rd.randint(0, 2 ** 128)))
-            setattr(self, "id", v)
-        return getattr(self, "id")
+    def _set_random_id(self):
+
+        if hasattr(self, "id"):
+            return getattr(self, "id")
```

Comment:
😅 I had this logic wrong at first - the whole point of this PR is to force folks to add this ID explicitly, so if it's already there we should skip the checks below! [Now that case is tested](https://github.com/plotly/dash/pull/1894/files#diff-16dfc841434d6081a8e87125cac6a04eb38359005a6d6b214f7409d750bf7878R533)